src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let firstCopyState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        firstCopyState = {
+          ...firstCopyState,
+          ...action.extraData,
+        };
+
+        break;
+
+      case 'removeProperties':
+        const secondCopyState = { ...firstCopyState };
```

Comment:
no need to create a separate variable here (and copy in the `addProperties` case) - you can mutate the `firstCopyState` object because you do the copy when pushing